sdv/multi_table/utils.py

```diff
@@ -0,0 +1,400 @@
+            Set of the tables that relationships will be removed.
+
+    Returns:
+        MultiTableMetadata:
+            Simplified metadata.
+    """
+    relationships = deepcopy(metadata.relationships)
+    for relationship in relationships:
+        parent_table = relationship['parent_table_name']
+        child_table = relationship['child_table_name']
+        is_parent_to_drop = parent_table in tables_to_drop
+        is_child_to_drop = child_table in tables_to_drop
+        if is_parent_to_drop or is_child_to_drop:
+            metadata.relationships.remove(relationship)
+            if is_parent_to_drop and parent_table in metadata.tables:
+                del metadata.tables[parent_table]
+            elif is_child_to_drop and child_table in metadata.tables:
+                del metadata.tables[child_table]
+
+    return metadata
+
+
+def _simplify_grandchildren(metadata, grandchildren):
+    """Simplify the grandchildren of the root table.
+
+    The logic to simplify the grandchildren is to:
+     - Drop all modelables columns.
+
+    Args:
+        metadata (MultiTableMetadata):
+            Metadata of the datasets.
+        grandchildren (set):
+            Set of the grandchildren of the root table.
+    """
+    for grandchild in grandchildren:
+        columns = metadata.tables[grandchild].columns
+        columns_to_drop = [
+            col_name for col_name in columns if columns[col_name]['sdtype'] in MODELABLE_SDTYPE
+        ]
+        for column in columns_to_drop:
+            del columns[column]
+
+    return metadata
+
+
+def _get_num_column_to_drop(metadata, child_table, max_col_per_relationships):
+    """Get the number of columns to drop from the child table.
+
+    Formula to determine how many columns to drop for a child
+        - n: number of columns of the table
+        - k: num_column_parameter. For beta's distribution k=4
+        - m: max_col_per_relationships
+
+        - minimum number of column to drop = n + k - sqrt(k^2 + 1 + 2m)
+
+    Args:
+        metadata (MultiTableMetadata):
+            Metadata of the datasets.
+        child_table (str):
+            Name of the child table.
+        max_col_per_relationships (int):
+            Maximum number of columns to model per relationship.
+
+    Returns:
+        tuple:
+            - num_cols_to_drop (int): Number of columns to drop from the child table.
+            - modelable_columns (dict): Dictionary that maps the modelable sdtype to their columns.
+    """
+    num_column_parameter = 4  # for beta distribution
+    columns = metadata.tables[child_table].columns
+    columns_to_sdtypes = {
+        column: columns[column]['sdtype'] for column in columns
+    }
+    sdtypes_to_columns = defaultdict(list)
+    for col, sdtype in columns_to_sdtypes.items():
+        sdtypes_to_columns[sdtype].append(col)
+
+    modelable_columns = {
+        key: value for key, value in sdtypes_to_columns.items() if key in MODELABLE_SDTYPE
+    }
```

Comment:
I think this can be simplified to something like:
```python
modelable_columns = defaultdict(list)
for col, col_meta in columns.items():
    if col_meta['sdtype'] in MODELABLE_SDTYPE:
        modelable_columns[col_meta['sdtype'].append(col)
```